library/swift/Headers.swift

```diff
@@ -4,45 +4,58 @@ import Foundation
 /// To instantiate new instances, see `{Request|Response}HeadersBuilder`.
 @objcMembers
 public class Headers: NSObject {
-  let headers: [String: [String]]
+  let container: HeadersContainer
 
   /// Get the value for the provided header name.
   ///
-  /// - parameter name: Header name for which to get the current value.
+  /// - parameter name: The case-insensitive header name for which to
+  ///                   get the current value.
+  ///
   ///
   /// - returns: The current headers specified for the provided name.
   public func value(forName name: String) -> [String]? {
-    return self.headers[name]
+    return self.container.value(forName: name)
   }
 
   /// Accessor for all underlying headers as a map.
```

Comment:
I'd add a `- warning:` to discourage using this dictionary to do equality key-based lookups, possibly linking to http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2 for context.

Reply:
added warning